graph-builder/src/plugins/openshift_secondary_metadata_parser/plugin.rs

```diff
@@ -0,0 +1,529 @@
+                    })
+                })
+                .is_some()
+        });
+        debug!(
+            "Sorted channels metadata of {} releases.",
+            sorted_releases.len()
+        );
+
+        Ok(io)
+    }
+}
+
+// Add some magic metadata which was manually added to quay
+fn add_manually_added_quay_metadata(graph: &mut cincinnati::Graph) -> Fallible<()> {
+    [
+        (
+            ("4.1.0-rc.7+amd64"),
+            vec![("io.openshift.upgrades.graph.previous.add", "4.1.0-rc.6")],
+        ),
+        (
+            ("4.1.0-rc.6+amd64"),
+            vec![
+                ("io.openshift.upgrades.graph.previous.add", "4.1.0-rc.5"),
+                // (
+                //     "io.openshift.upgrades.graph.release.channels",
+                //     "prerelease-4.1",
+                // ),
+            ],
+        ),
+        (
+            ("4.1.0-rc.8+amd64"),
+            vec![
+                ("io.openshift.upgrades.graph.previous.add", "4.1.0-rc.7"),
+                // (
+                //     "io.openshift.upgrades.graph.release.channels",
+                //     "prerelease-4.1",
+                // ),
+                //     "description": "",
+                //     "io.openshift.upgrades.graph.release.manifestref": "sha256:8250bbf79d4f567e24a7b85795aba97f9e75a9df18738891a0cb6ba42e422584",
+                //     "url": "https://access.redhat.com/errata/RHBA-2019:0758-09",
+                //     "io.openshift.upgrades.graph.release.arch": "amd64",
+            ],
+        ),
+        (
+            ("4.1.0-rc.9+amd64"),
+            vec![
+                (
+                    "io.openshift.upgrades.graph.previous.add",
+                    "4.1.0-rc.7,4.1.0-rc.8",
+                ),
+                // (
+                //     "io.openshift.upgrades.graph.release.channels",
+                //     "prerelease-4.1",
+                // ),
+                //     "description": "",
+                //     "io.openshift.upgrades.graph.release.arch": "amd64",
+                //     "io.openshift.upgrades.graph.release.manifestref": "sha256:49c4b6bf70061e522e3525aed534d087c9abfba7c39cbcbdd1bd770ab096bf9e",
+                //     "url": "https://access.redhat.com/errata/RHBA-2019:0758-09"
+            ],
+        ),
+        (
+            ("4.1.0+amd64"),
+            vec![
+                (
+                    "io.openshift.upgrades.graph.previous.add",
+                    "4.1.0-rc.4,4.1.0-rc.9",
+                ),
+                // (
+                //     "io.openshift.upgrades.graph.release.channels",
+                //     "prerelease-4.1,stable-4.1",
+                // ),
+            ],
+        ),
+        (
+            ("4.2.13-s390x+s390x"),
+            vec![
+                (
+                    "io.openshift.upgrades.graph.previous.add",
+                    "4.2.10-s390x,4.2.12-s390x",
+                ),
+                ("io.openshift.upgrades.graph.previous.remove", "*"),
```

Comment:
This can go into cincinnati-graph-data

Reply:
Wait, this [is][1] in cincinnati-graph-data.  I don't think you need to manually inject it; might be a bug somewhere in the Rust parser?

[1]: https://github.com/openshift/cincinnati-graph-data/blob/b70334977789366606fcec18ab4207870c6df279/blocked-edges/4.2.13-s390x.yaml#L3

Reply:
It is, and I think it is better to remove it OK this additional metadata, but it needs a workaround. The workaround means to have the EdgeRemoveAdd plugin remove all it's processed metadata. This is required because this parser adds one `remove_regex` metadata, when the quay labels have this expanded to one `remove` metadata entry per match, so they won't be compatible.